libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py

```diff
@@ -1718,6 +1718,12 @@ def write_UART_config(self, f):
                 f.write(
                     '#define HAL_UART_IO_DRIVER constexpr ChibiOS::UARTDriver &uart_io = serial%sDriver;\n' % (index)
                 )
+                f.write(
```

Comment:
This makes
```
libraries/AP_Vehicle/AP_Vehicle.cpp:361:        serial_manager.set_protocol_and_baud(HAL_UART_IOMCU_IDX, AP_SerialManager::SerialProtocol_IOMCU, 0);
```

redundant - I think we should remove the code in AP_Vehicle.

I think it would be nice to do away with `HAL_UART_IOMCU_IDX` entirely.  I'd really like to make IOMCU "just another serial protocol"...

Reply:
It seems dumb to expose the parameters tho. Under what circumstances would the user ever change them (except CubeRed).

Reply:
Again this is not correct for CubeRed. please check above comment for this. The parameter will no longer be exposed if not desired, fix for that is here https://github.com/ArduPilot/ardupilot/pull/29493

Reply:
removed